frameworks/SapientML/exec.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os
+import tempfile as tmp
+
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from frameworks.shared.callee import call_run, result
+from frameworks.shared.utils import Timer
+from sapientml import SapientML
+from sapientml.util.logging import setup_logger
+from sklearn.preprocessing import OneHotEncoder
+
+os.environ["JOBLIB_TEMP_FOLDER"] = tmp.gettempdir()
+os.environ["OMP_NUM_THREADS"] = "1"
+os.environ["OPENBLAS_NUM_THREADS"] = "1"
+os.environ["MKL_NUM_THREADS"] = "1"
+
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    import re
+
+    import pandas as pd
+
+    log.info(f"\n**** Sapientml ****\n")
+
+    is_classification = config.type == "classification"
+    is_multiclass = dataset.problem_type = "multiclass"
+    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith("_")}
+
+    train_path, test_path = dataset.train.path, dataset.test.path
+    target_col = dataset.target.name
+
+    # Read parquet using pandas
+    X_train = pd.read_csv(train_path)
+    X_test = pd.read_csv(test_path)
+
+    # Removing unwanted sybols from column names (exception case)
+    X_train.columns = [re.sub("[^A-Za-z0-9_.]+", "", col) for col in X_train.columns]
+    X_test.columns = [re.sub("[^A-Za-z0-9_.]+", "", col) for col in X_test.columns]
+    target_col = re.sub("[^A-Za-z0-9_.]+", "", target_col)
+
+    # y_train and y_test
+    y_train = X_train[target_col].reset_index(drop=True)
+    y_test = X_test[target_col].reset_index(drop=True)
+
+    # Drop target col from X_test
+    X_test.drop([target_col], axis=1, inplace=True)
+
+    # Sapientml
+    output_dir = config.output_dir + "/" + "outputs" + "/" + config.name + "/" + str(config.fold)
+    predictor = SapientML([target_col], task_type="classification" if is_classification else "regression")
```

Comment:
From the abstract, it seems that there is meta-learning involved. Are there datasets in the meta-learning corpus that are also in the AutoML benchmark? If so, is there a way to avoid "turn off" the inclusion of that data from the meta-model for individual evaluations (e.g., don't use meta-information found on the Santander dataset while evaluating on the Santander dataset?).

Reply:
No, the meta-learning corpus does not include no dataset included in the AutoML benchmark.